src/components/button/index.js

```diff
@@ -2,21 +2,28 @@ import React from 'react';
 
 import './styles.scss';
 
-const Button = ({ icon, title, iconStyle, style, indicator, onClick, small }) => {
+const Button = ({ icon, title, iconStyle, style, indicator, onClick, onKeyDown, small }) => {
 
   function handleClick() {
     if(typeof onClick === 'function') {
       onClick();
     }
   }
 
+  function handleKeyDown(event) {
+    if(typeof onKeyDown === 'function') {
```

Comment:
What is the reason for checking whether `onKeyDown` is a function? I see un wanting to check it it exists, but assuming it does exist, what other potential types might there be? If none, then perhaps something like this:

`onKeyDown && onKeyDown(event);`

Reply:
I'll admit I blindly followed the pattern of the `handleClick` before it.

`onKeyDown && onKeyDown(event);`
But reviewing it, there's a particular case where short-circuiting statement can produce errors.

What if, for some reason, somebody passed something like
```jsx
<input
  value={this.state.value}
  onKeyDown // i.e. boolean true, huh? what??? where's the function?????
/>
```

The first clause is just an `onKeyDown` existence check, right?
Now, try the second clause, where `onKeyDown === true`, and I mean a **boolean** `true`.

```js
let onKeyDown = true;
let event = 123;
onKeyDown && onKeyDown(123); // Uncaught TypeError: onKeyDown is not a function
```

There's brevity... and there's safety.

Reply:
I still don't think this is the right way of going about it.

Technically you are right and someone might assign some non-function value to the prop, but by the same token someone could put an infinite loop into the function, so something else that will equally break the code. In this case, since we are not exposing this to any outside consumers, it's our job to make sure we don't do anything like that. The reason I bring this up is that doing this validation opens the door to doing similar validations on all props, which will in turn overwhelm us.

One solution is to use Flow or, better yet, PropTypes — which come with React for free.